backend/app/Models/Tag.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\SoftDeletes;
+
+class Tag extends Model
+{
+    use HasFactory, SoftDeletes;
+
+    protected $guarded = [];
+
+    public function users()
+    {
+        return $this->belongsToMany(User::class, 'user_tag');
```

Comment:
update we don't need to specify the pivot table if table naming is correct.
```suggestion
        return $this->belongsToMany(User::class);
```